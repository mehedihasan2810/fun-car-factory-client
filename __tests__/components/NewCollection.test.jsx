import NewCollection from "../../src/components/NewCollention/NewCollention";
import {
  describe,
  expect,
  renderWithRouter,
  rendererWithRouter,
  screen,
  test,
  toJson,
} from "../utils/utils";

describe("<NewCollection />", () => {
  test("Should render succesfully and there should be a heading, button and two img, ", () => {
    renderWithRouter(<NewCollection />);

    const heading = screen.getByText((content, element) => {
      return (
        element.tagName.toLowerCase() === "h4" &&
        content.startsWith("Brand New Toys")
      );
    });
    const button = screen.getByText("Explore");
    const imgs = screen.getAllByRole("img");

    expect(heading).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    expect(imgs.length).toBe(2);
  });

  test("should match snapshot", () => {
    const component = rendererWithRouter(<NewCollection />);
    let tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
