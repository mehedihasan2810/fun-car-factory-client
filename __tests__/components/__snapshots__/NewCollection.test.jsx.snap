// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NewCollection /> > should match snapshot 1`] = `
<div
  className="new-collection-wrapper"
>
  <div
    className="new-collection-left-gradient"
  />
  <div
    className="new-collection-right-gradient"
  />
  <section
    className="new-collection-container parallax"
    data-speed="25"
  >
    <div
      className="new-collection-images-container"
    >
      <img
        alt=""
        src="https://images.unsplash.com/photo-1511415518647-9e5da4fd803f?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=2070&q=80"
      />
      <img
        alt=""
        src="https://images.unsplash.com/photo-1519283053578-3efb9d2e71bd?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=2070&q=80"
      />
    </div>
    <div
      className="new-collection-info-container"
    >
      <p>
        New Collection
      </p>
      <h4>
        Brand New Toys 
        <br />
         Just Arrived
      </h4>
      <a
        className="btn-primary"
        href="/"
        onClick={[Function]}
      >
        Explore
      </a>
    </div>
  </section>
</div>
`;
